client/app/dbaas/logs/detail/streams/alerts/add/alerts-add.controller.js

```diff
@@ -1,9 +1,48 @@
-class LogsStreamsAlertsAddCtrl {
-    constructor ($stateParams) {
-        this.$stateParams = $stateParams;
-        this.serviceName = this.$stateParams.serviceName;
-        this.stream = this.$stateParams.streamId;
+class AlertsAddCtrl {
+    constructor ($q, $state, $stateParams, $window, CloudMessage, ControllerHelper, StreamsAlertsService) {
+        this.$q = $q;
+        this.$state = $state;
+        this.serviceName = $stateParams.serviceName;
+        this.streamId = $stateParams.streamId;
+        this.alertType = $stateParams.type;
+        this.$window = $window;
+        this.CloudMessage = CloudMessage;
+        this.ControllerHelper = ControllerHelper;
+        this.StreamsAlertsService = StreamsAlertsService;
+    }
+
+    $onInit () {
+        this.alert = this.StreamsAlertsService.getNewAlert(this.alertType);
+    }
+
+    /**
+     * Adds a new alert by making an API call
+     *
+     * @memberof AlertsAddCtrl
+     */
+    addAlert () {
+        if (this.form.$invalid) {
+            return this.$q.reject();
+        }
+
+        this.CloudMessage.flushChildMessage();
+        this.addingAlert = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () =>
+                this.StreamsAlertsService.addAlert(this.serviceName, this.streamId, this.alert)
+                    .then(() => this.$state.go("dbaas.logs.detail.streams.alerts"))
```

Comment:
CloudNavigation.previousState.go();

In the main Controller of logs (the controller associated with dbaas.logs.detail) => 

this.CloudNavigation.init({
            state: "network.iplb.detail",
            stateParams: {
                serviceName: this.serviceName
            }
        });

or $window.history.back()... however if the user access the page directly it won't be very user friendly for him

Reply:
I have changed it to use $window.history.back() :)